### PR TITLE
Revise cmake patch

### PR DESCRIPTION
**Pull request by dg0yt**
This is the simplified restbed patch as discussed in the vcpkg PR. I admit there is no significant gain. So feel free to reject it.

### CODE REVIEW

File: ports/restbed/fix-cmake.patch [hunk 1]
```diff
@@ -24,40 +21,29 @@ diff --git a/CMakeLists.txt b/CMakeLists.txt
...
+@@ -93,8 +96,13 @@ else ( )
  	set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME} )
  endif ( )	
  set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR} VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR} )
++set ( LIBRARY_NAME "${SHARED_LIBRARY_NAME}")
 +endif ( )
  
  if ( BUILD_SSL )
-+if ( BUILD_SHARED_LIBS )
++    target_link_libraries( ${LIBRARY_NAME} LINK_PRIVATE OpenSSL::SSL)
```
Comment (chausner) [inline ports/restbed/fix-cmake.patch:37]: The original code also linked to crypto_LIBRARY_SHARED/crypto_LIBRARY_STATIC which seem to map to OpenSSL::Crypto. Are you sure we don't need to link to it? Also, what about CMAKE_DL_LIBS that the original code links to in the static library case?

File: ports/restbed/fix-cmake.patch [hunk 2]
```diff
@@ -24,40 +21,29 @@ diff --git a/CMakeLists.txt b/CMakeLists.txt
...
++elseif ( 1 )
```
Comment (chausner) [inline ports/restbed/fix-cmake.patch:38]:
```suggestion
```

**Review by chausner — CHANGES_REQUESTED** [review]

File: ports/restbed/fix-cmake.patch (hunk 1 above)
Comment (dg0yt) [reply ports/restbed/fix-cmake.patch:37]: I compiled at least on one static triplet before submitting.
I assume crypto_LIBRARY_* and CMAKE_DL_LIBS are just transitive usage requirements of the main openssl lib.
Target OpenSSL::SSL depends on OpenSSL::Crypto and on whatever CMAKE_DL_LIBS actually is. That's the key benefit of imported targets if they are done correctly: Consumers (here: restbed) don't need to know and repeat the transitive usage requirements.
Things look different if restbed actually used libcrypto or dl directly.

File: ports/restbed/fix-cmake.patch (hunk 2 above)
Comment (dg0yt) [reply ports/restbed/fix-cmake.patch:38]: This line is intentional to catch all other cases.
It is a minimal patch, not the most beautiful code ;-)

Comment (chausner) [reply ports/restbed/fix-cmake.patch:38]: Ah, I missed the `else` further down. To be honest, I think I would leave the patch as it is. It is certainly not as concise as your suggestion but I would rather go for readability in this case than minimizing the patch.

Comment (dg0yt) [reply ports/restbed/fix-cmake.patch:38]: Fine for me.

**Pull request closed without merging**